### PR TITLE
OPT-26580 DC0994_US | Frictionless | Add sticky banner Upload CTA

### PR DESCRIPTION
**Pull request by ivanvatadobe**
POC for addition to sticky banner on frictionless pages. Mimicking behavior of a button that can interact and start the widget. Using javascript function to achieve the simulated click, but is not the plan for the final feature. The final feature will likely be built by the eng team that builds the frictionless widgets. 

Resolves: [OPT-26580](https://jira.corp.adobe.com/browse/OPT-26580)

**Test URLs for psi check:**
- Before: https://main--milo--adobecom.hlx.page/?martech=off
- After: https://dc0944stickybanner--milo--adobecom.hlx.page/?martech=off

**Test URLs to view feature:**
- Before: https://main--dc--adobecom.hlx.page/drafts/viloria/fragments/dc0994/testercompresspage?mep=%2Fdc-shared%2Ffragments%2Ftests%2F2024%2Fq2%2Fdc0994%2Fdc0994.json--target-var1-addstickybanner&mepButton=off
- After: https://main--dc--adobecom.hlx.page/drafts/viloria/fragments/dc0994/testercompresspage?mep=%2Fdc-shared%2Ffragments%2Ftests%2F2024%2Fq2%2Fdc0994%2Fdc0994.json--target-var1-addstickybanner&milolibs=dc0944stickybanner&mepButton=off

Note: 
- This ticket marked high priority because it is a blocker for test launch. 
-  The branch name is dc0944stickybanner--- this was a typo when the branch was created. The test this code is being written for is called 'DC0994'. I have updated the urls above with the proper naming conventions.

### CODE REVIEW

File: libs/mep/dc0944/aside/aside.js [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+  const ctas = el.querySelectorAll('a');
+  ctas.forEach((cta) => {
+    if (cta.href.toLowerCase().includes('#module')) {
+      cta.addEventListener('click', (e) => {
+        e.preventDefault();
+        const moduleCta = document.querySelectorAll('.CTAButtons');
+        window.scrollTo(0, 0);
+        if (moduleCta.length) {
+          moduleCta[0].click();
+        }
```
Comment (vgoodric) [inline libs/mep/dc0944/aside/aside.js:59]:
```suggestion
        window.scrollTo(0, 0);
        document.querySelector('.CTAButtons')?.click();
```

Comment (vgoodric) [reply libs/mep/dc0944/aside/aside.js:59]: Think this can be reduced to less lines.

**Comment by vgoodric** [comment]
For CSS, switch to using `.aside.frictionless` instead of `.section.hide-on-step-start` unless needed.

**Review by vgoodric — CHANGES_REQUESTED** [review]
Please see comments

File: libs/mep/dc0944/aside/aside.js [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+export default function init(el) {
+  el.classList.add('con-block');
+  const blockData = getBlockData(el);
+  const blockText = decorateLayout(el);
+  decorateBlockText(blockText, blockData);
+  decorateStaticLinks(el);
+  formatPromoButton(el);
+  if (el.classList.contains('frictionless')) {
+    const moduleCtas = document.querySelectorAll('.CTAButtons');
+    if (moduleCtas.length) {
```
Comment (robert-bogos) [inline libs/mep/dc0944/aside/aside.js:239]: Nit: Same here, we can skip the caching
```suggestion
    if (document.querySelectorAll('.CTAButtons').length) {
```

File: libs/mep/dc0944/aside/aside.js [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+function addPromobar(sourceEl, parent) {
+  const newPromo = sourceEl.cloneNode(true);
+  parent.appendChild(newPromo);
+}
+
+function checkViewportPromobar(foreground) {
+  const { children, childElementCount: childCount } = foreground;
+  if (childCount < 2) addPromobar(children[childCount - 1], foreground);
+  if (childCount < 3) addPromobar(children[childCount - 1], foreground);
+}
```
[the remark this answers is not in the document]
Comment (ivanvatadobe) [reply libs/mep/dc0944/aside/aside.js:125]: Although these files look like they are completely new code, they're actually just experimental/temporary copies for an upcoming test. 

I do not feel comfortable committing this change as I did not change the original for my test version.

File: libs/mep/dc0944/aside/aside.js [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+    decorateIconStack(el);
+  }
+  return foreground;
+}
+
+export default function init(el) {
+  el.classList.add('con-block');
+  const blockData = getBlockData(el);
+  const blockText = decorateLayout(el);
+  decorateBlockText(blockText, blockData);
```
[the remark this answers is not in the document]
Comment (ivanvatadobe) [reply libs/mep/dc0944/aside/aside.js:230]: I do not feel comfortable committing this change as I did not change the original for my test version.

File: libs/mep/dc0944/aside/aside.js [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+                      <circle cx="10" cy="10" r="10" transform="translate(10500 -3403)" fill="#707070"></circle>
+                      <line y1="8" x2="8" transform="translate(10506 -3397)" fill="none" stroke="#fff" stroke-width="2"></line>
+                      <line x1="8" y1="8" transform="translate(10506 -3397)" fill="none" stroke="#fff" stroke-width="2"></line>
+                    </g>
+                  </svg>`;
+
+function linkSectionToModuleStep(el) {
+  const parentSection = el.closest('.section');
+  parentSection.classList.add('hide-on-step-start');
+}
```
[the remark this answers is not in the document]
Comment (ivanvatadobe) [reply libs/mep/dc0944/aside/aside.js]: Change added.

**Comment by ivanvatadobe** [comment]
@vgoodric I have made a single CSS update as a response to QA testing on milo preview pages. 

The changes are committed and pushed. There are no other requests from QA.

**Comment by rgclayton** [comment]
Why is this being submitted to the Milo repo? Shouldn't this POC code live under DC's repo?

cc: @vgoodric

**Review by vgoodric — APPROVED** [review]

**Comment by vgoodric** [comment]
> POC code live under DC

@rgclayton This on the Milo repo because it's using a version of aside for the test.  It's quarantined to the MEP folder, so it doesn't really matter which repo it's on and it's easier to use the same repo that the block you're copying is also on.  Less updates to the code.

**Review by denlight — APPROVED** [review]

**Pull request merged**